### ISSUE TITLE
python:3.11-slim-bookworm CVE-2023-5752

### ISSUE THREAD

**Issue by trottomv** [docker-library/python#889]
| Library        | Vulnerability  | Severity | Status | Installed Version | Fixed Version | Title                                                        |
|----------------|----------------|----------|--------|---------------------|---------------|--------------------------------------------------------------|
| pip (METADATA) | CVE-2023-5752 | MEDIUM   | fixed  | 23.2.1             | 23.3          | pip: Mercurial configuration injectable in repo revision when installing via pip [Link](https://avd.aquasec.com/nvd/cve-2023-5752) |

**Comment by LaurentGoderre** [comment]
Need to be updated here: https://github.com/python/cpython/blob/b6535ea7ecf40c51a9a13089d961c29abeaffa5e/Lib/ensurepip/__init__.py#L15

**Comment by LaurentGoderre** [comment]
https://github.com/python/cpython/pull/112710

**Comment by trottomv** [comment]
Hi @LaurentGoderre 

Is it not necessary to modify the pip version here as well?
https://github.com/docker-library/python/blob/8bc80d1109001365559eded16423ba3692eff1ff/3.11/slim-bookworm/Dockerfile#L137

(and in the "not slim" bookworm also)
https://github.com/docker-library/python/blob/8bc80d1109001365559eded16423ba3692eff1ff/3.11/bookworm/Dockerfile#L103

**Comment by LaurentGoderre** [comment]
@trottomv that version is derived from the location I pointed to in the update script.

**Comment by rv0lt** [comment]
As far as I am aware, that issue is only going to be fixed in python 3.13 (currently in alpha). The maintainers decided against back porting to previous versions

https://github.com/python/cpython/pull/112719

**Comment by edmorley** [comment]
As filed, this issue is now resolved since the Python 3.11 images now ship with pip 24.0:
https://github.com/docker-library/python/blob/cc2cf19f2c9a440706d17b5937bf104052eef967/versions.json#L24-L32

```
$ docker run -q --rm python:3.11-slim-bookworm pip --version
pip 24.0 from /usr/local/lib/python3.11/site-packages/pip (python 3.11)
```

Images for older Python versions do still include older pip, but that's by design (since these Docker images intentionally ship with the same pip version that is bundled with the `ensurepip` module for that Python version).

As such, this issue can be closed.
